src/app/static/src/app/store/baseline/actions.ts

```diff
@@ -1,33 +1,15 @@
-import {ActionContext, ActionPayload, ActionTree} from 'vuex';
+import {ActionContext, ActionTree, Payload} from 'vuex';
 import {BaselineState} from "./baseline";
 import {RootState} from "../../main";
 import {api} from "../../apiService";
-
-export interface PJNZ {
-    filename: string
-    country: string
-}
-
-export interface BaselineData {
-    pjnz?: PJNZ
-}
+import {PjnzResponse} from "../../generated";
+import {BaselineData} from "../../types";
 
 export type BaselineActionTypes = "PJNZUploaded" | "PJNZUploadError" | "BaselineDataLoaded"
 
-export interface BaselinePayload extends ActionPayload {
+export interface BaselinePayload<T> extends Payload {
     type: BaselineActionTypes
-}
-
-export interface BaselineDataLoaded extends BaselinePayload {
-    payload: BaselineData
-}
-
-export interface PJNZUploaded extends BaselinePayload {
-    payload: PJNZ
-}
-
-export interface PJNZUploadError extends BaselinePayload {
-    payload: string
+    payload: T
```

Comment:
I realised this is what we really want from the type-checking here - to constrain the "type" to being a valid "BaselineActionTypes" and specify what the payload is on an action by action basis.